code/datums/ai/basic_mobs/targetting_datums/with_object.dm

```diff
@@ -0,0 +1,34 @@
+/**
+ * Find mobs who are holding the configurable object type
+ *
+ * This is an extension of basic targeting behaviour, that allows you to
+ * only target the mob if they have a specific item in their hand.
+ *
+ */
+/datum/targetting_datum/basic/holding_object
+	// We will find mobs who are holding this object in their hands
+	var/object_type_path = null
+
+/**
+ * Create an instance of the holding object targeting datum
+ *
+ * * object_type_path Pass an object type path, this will be compared to the items
+ *   in targets hands to filter the target list.
+ */
+/datum/targetting_datum/basic/holding_object/New(object_type_path)
+	if (!ispath(object_type_path))
+		stack_trace("trying to create an item targeting datum with no valid typepath")
+		// Leaving object type as null will make this basically a noop
+		return
+	src.object_type_path = object_type_path
+
+///Returns true or false depending on if the target can be attacked by the mob
+/datum/targetting_datum/basic/holding_object/can_attack(mob/living/living_mob, atom/target, vision_range)
+	if (object_type_path == null)
+		return FALSE // no op
+	// Check if our parent behaviour agrees we can attack this target, ignoring factions
+	var/can_attack = ..(living_mob, target, vision_range, FALSE)
```

Comment:
The `var/can_attack = ..(living_mob, target, vision_range, FALSE)` could potentially cause issues if the parameters ever change. 

Are you willing to change the proc params to `/datum/targetting_datum/basic/holding_object/can_attack(mob/living/living_mob, atom/target, vision_range, check_faction = FALSE)` and then just call parent without args?

Not a blocking suggestion, just a stylistic one.

Reply:
it's cringe and gross but okay.

Reply:
o7